src/utils/CometAdapter.cpp

```diff
@@ -0,0 +1,589 @@
+            <td VALIGN="middle" ALIGN = "center" ROWSPAN=1> any signal-/preprocessing tool @n (in mzML format)</td>
+            <td VALIGN="middle" ALIGN = "center" ROWSPAN=1> @ref TOPP_IDFilter or @n any protein/peptide processing tool</td>
+        </tr>
+    </table>
+</CENTER>
+
+    @em Comet must be installed before this wrapper can be used. This wrapper
+    has been successfully tested with version 2016.01.2 of Comet.
+
+    Comet settings not exposed by this adapter can be directly adjusted using a param file, which can be generated using comet -p.
+    By default, All (!) parameters available explicitly via this param file will take precedence over the wrapper parameters.
+
+    Parameter names have been changed to match names found in other search engine adapters, however some are Comet specific.
+    For a detailed description of all available parameters check the Comet documentation at http://comet-ms.sourceforge.net/parameters/parameters_201601/
+
+    <B>The command line parameters of this tool are:</B>
+    @verbinclude TOPP_CometAdapter.cli
+    <B>INI file documentation of this tool:</B>
+    @htmlinclude TOPP_CometAdapter.html
+*/
+
+// We do not want this class to show up in the docu:
+/// @cond TOPPCLASSES
+
+
+class TOPPCometAdapter :
+  public TOPPBase
+{
+public:
+  TOPPCometAdapter() :
+    TOPPBase("CometAdapter", "Annotates MS/MS spectra using Comet.")
+  {
+  }
+
+protected:
+  void registerOptionsAndFlags_()
+  {
+
+    registerInputFile_("in", "<file>", "", "Input file");
+    setValidFormats_("in", ListUtils::create<String>("mzML"));
+    registerOutputFile_("out", "<file>", "", "Output file");
+    setValidFormats_("out", ListUtils::create<String>("idXML"));
+    registerInputFile_("database", "<file>", "", "FASTA file", true, false, ListUtils::create<String>("skipexists"));
+    setValidFormats_("database", ListUtils::create<String>("FASTA"));
+    registerInputFile_("comet_executable", "<executable>",
+      // choose the default value according to the platform where it will be executed
+      "comet.exe",
+      "Comet executable of the installation e.g. 'comet.exe'", true, false, ListUtils::create<String>("skipexists"));
+
+    //
+    // Optional parameters //
+    //
+
+    //Files
+    registerOutputFile_("pin_out", "<file>", "", "Output file - for Percolator input", false);
+    setValidFormats_("pin_out", ListUtils::create<String>("pin"));
+    registerInputFile_("default_params_file", "<file>", "", "Default Comet params file. All parameters of this take precedence. A template file can be generated using comet.exe -p", false, false, ListUtils::create<String>("skipexists"));
+    setValidFormats_("default_params_file", ListUtils::create<String>("txt"));
+    //registerIntOption_("threads", "<num>", 1, "number of threads", false, true);
+
+    //Masses
+    registerDoubleOption_("precursor_mass_tolerance", "<tolerance>", 10.0, "Precursor mass tolerance, peptide_mass_tolerance(Comet)", false, false);
```

Comment:
where / how did you choose these default values? I suggest to use these from comet itself, e.g. http://comet-ms.sourceforge.net/parameters/parameters_201601/comet.params.high-high